### DISCUSSION TITLE
Use both `alker0/chezmoi.vim` and nvim-treesitter?

### DISCUSSION THREAD

**Discussion by xfzv** [nvim-treesitter/nvim-treesitter#3810] [General]
### Describe the bug

TL;DR: I'd like to use both [alker0/chezmoi.vim](https://github.com/alker0/chezmoi.vim) and nvim-treesitter without breaking syntax highlighting for `chezmoi` specific files.

Related issue: https://github.com/alker0/chezmoi.vim/issues/45

Enabling both plug-ins breaks the syntax highlighting for many of the specific `chezmoi` files like templates or `.chezmoiexternal.$FORMAT`.

Is there any way to achieve this? maybe with

https://github.com/nvim-treesitter/nvim-treesitter/blob/2072692aaa4b6da7c354e66c2caf4b0a8f736858/README.md?plain=1#L130-L138

My idea is to disable tree-sitter only if the current filetype matches  `*.chezmoitmpl` but I'm not skilled enough to come up with such a function.

Any help is appreciated.

Thanks.

### To Reproduce

1. Load both [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) and [alker0/chezmoi.vim](https://github.com/alker0/chezmoi.vim) with Packer
2. Open any `chezmoi` specific file like  `.chezmoi.yml.tmpl` which contains both yaml and go-template content
3. Syntax highlighting is broken:
![image](https://user-images.githubusercontent.com/78810647/201986679-1c984c34-f4b7-4bd1-8b67-55e6c6bf1a43.png)

5. Disable tree-sitter and run `:PackerCompile`
6. Open `.chezmoi.yml.tmpl` again
7. Syntax highlighting is correct:

![image](https://user-images.githubusercontent.com/78810647/201986793-92d7f7ea-94e6-4db0-ab6a-63065e59d9f5.png)

### Output of `:checkhealth nvim-treesitter`

```markdown
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found 0.20.7 (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v19.1.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: cc (GCC) 12.2.0
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - rasi           ✓ ✓ ✓ ✓ .
  - norg_meta      ✓ . . ✓ .
  - http           ✓ . . . ✓
  - go             ✓ ✓ ✓ ✓ ✓
  - diff           ✓ . . . .
  - kotlin         ✓ ✓ ✓ . ✓
  - json           ✓ ✓ ✓ ✓ .
  - vim            ✓ ✓ ✓ . ✓
  - lua            ✓ ✓ ✓ ✓ ✓
  - markdown_inline✓ . . . ✓
  - bash           ✓ ✓ ✓ . ✓
  - toml           ✓ ✓ ✓ ✓ ✓
  - python         ✓ ✓ ✓ ✓ ✓
  - cpp            ✓ ✓ ✓ ✓ ✓
  - query          ✓ ✓ ✓ ✓ ✓
  - jsonc          ✓ ✓ ✓ ✓ ✓
  - yaml           ✓ ✓ ✓ ✓ ✓
  - rust           ✓ ✓ ✓ ✓ ✓
  - comment        ✓ . . . .
  - c              ✓ ✓ ✓ ✓ ✓
  - help           ✓ . . . .
  - typescript     ✓ ✓ ✓ ✓ ✓
  - html           ✓ ✓ ✓ ✓ ✓
  - scss           ✓ . . ✓ .
  - cmake          ✓ . ✓ . .
  - gitignore      ✓ . . . .
  - make           ✓ . ✓ . ✓
  - css            ✓ . ✓ ✓ ✓
  - norg           ✓ . ✓ . ✓
  - markdown       ✓ . ✓ . ✓
  - javascript     ✓ ✓ ✓ ✓ ✓

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}
```

### Output of `nvim --version`

```text
NVIM v0.8.1
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by builduser

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```
```


### Additional context

_No response_

**Comment by clason** [comment]
You can either add `additional_vim_regex_highlighting = true` to your nvim-treesitter config, or add a `BufRead` etc. autocommand that calls `vim.treesitter.stop()` on `*.chezmoitmpl` files.

**Comment by alker0** [comment]
This may work if for disabling only highlighting:
```lua
highlight = {
  disable = function(lang, buf)
    -- check if 'b:current_syntax' includes 'chezmoitmpl'
    if string.find(vim.b.current_syntax, 'chezmoitmpl') then
      return true
    end
  end,
  -- additional_vim_regex_highlighting = true, <- not required
}
```

**Reply by xfzv** [reply]
> This may work if for disabling only highlighting:
> 
> ```lua
> highlight = {
>   disable = function(lang, buf)
>     -- check if 'b:current_syntax' includes 'chezmoitmpl'
>     if string.find(vim.b.current_syntax, 'chezmoitmpl') then
>       return true
>     end
>   end,
>   -- additional_vim_regex_highlighting = true, <- not required
> }
> ```

This works but I'm getting the following error when opening any file:

```viml
Error detected while processing BufReadPost Autocommands for "\V/home/xfzv/dotfiles/\*"..function chezmoi#filetype#handle_c
hezmoi_filetype[44]..<SNR>78_handle_source_file[17]..<SNR>78_run_default_detect[6]..BufReadPost Autocommands for "*.yml"..Fi
leType Autocommands for "*"..function matchup#loader#init_buffer[15]..matchup#ts_engine#is_hl_enabled[4]..<SNR>85_forward:
line    1:
E5108: Error executing lua /home/xfzv/.config/nvim/lua/plugins/treesitter.lua:19: bad argument #1 to 'find' (string expecte
d, got nil)
stack traceback:
        [C]: in function 'find'
        /home/xfzv/.config/nvim/lua/plugins/treesitter.lua:19: in function 'disable'
        ...er/start/nvim-treesitter/lua/nvim-treesitter/configs.lua:391: in function 'is_hl_enabled'
        [string "luaeval()"]:1: in main chunk
Press ENTER or type command to continue
```

I ended up using:

```lua
highlight = {
  enable = true,
  disable = function(lang, bufnr)
    if string.find(vim.bo.filetype, "chezmoitmpl") then
      return true
    end
  end,
},
```

> -- additional_vim_regex_highlighting = true, <- not required

Indeed, it does not make any difference.

Thank you!
